ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -89,45 +85,24 @@ public DdlCommandResult run(MetaStore metaStore, boolean isValidatePhase) {
 
   private void deleteTopicIfNeeded(StructuredDataSource dataSource, boolean isValidatePhase) {
     if (!isValidatePhase && deleteTopic) {
-
-      executeWithRetries(new Callable<Void>() {
-        @Override
-        public Void call() throws Exception {
-          kafkaTopicClient.deleteTopics(
-              Collections.singletonList(dataSource.getKsqlTopic().getKafkaTopicName()));
-          return null;
-        }
-      }, "Could not delete the corresponding kafka topic: "
-           + dataSource.getKsqlTopic().getKafkaTopicName());
-
-      if (dataSource.getKsqlTopic().getKsqlTopicSerDe().getSerDe()
-          == DataSource.DataSourceSerDe.AVRO) {
-        executeWithRetries(new Callable<Void>() {
-          @Override
-          public Void call() throws Exception {
-            schemaRegistryClient
-                .deleteSubject(sourceName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
-            return null;
-          }
-        }, "Could not clean up the schema registry for topic: " + sourceName);
-      }
-    }
-  }
-
-  private void executeWithRetries(Callable<Void> callable, String errorMessage) {
-    int retries = 0;
-    while (retries < NUM_RETRIES) {
       try {
-        if (retries != 0) {
-          Thread.sleep(RETRY_BACKOFF_MS);
-        }
-        callable.call();
-        break;
+        ExecutorWithRetries.execute(
+            () -> kafkaTopicClient.deleteTopics(
+                    Collections.singletonList(
+                        dataSource.getKsqlTopic().getKafkaTopicName())),
+            ExecutorWithRetries.RetryBehaviour.ALWAYS);
       } catch (Exception e) {
-        retries++;
-      } finally {
-        if (retries == NUM_RETRIES) {
-          throw new KsqlException(errorMessage);
+        throw new RuntimeException(e);
```

Comment:
Can you add the error message back in, and maintain the same exception type, please:

```
  throw new KsqlException("Could not delete the corresponding kafka topic: "  + dataSource.getKsqlTopic().getKafkaTopicName(), e);
```